### PR TITLE
Scan and parse identifier token

### PR DESCRIPTION
**Pull request by leewei05**
I think it would be handy if we introduce identifier before implementing statements.

### CODE REVIEW

File: lexer.l [hunk 1]
```diff
@@ -22,6 +22,9 @@
 #define LINENO (yylineno - 1)
 %}
 
+/* C99 7.1.3: All identifiers that begin with an underscore and either an uppercase letter or another
```
Comment (Lai-YT) [inline lexer.l:25]: I suggest that we have our line length under 80 characters 😀 
See [Google C++ Style Guide: Line Length](https://google.github.io/styleguide/cppguide.html#Line_Length) for the details!

File: ast.cpp [hunk 2]
```diff
@@ -55,6 +55,18 @@ class ProgramNode : public AstNode {
   std::vector<std::unique_ptr<ExprNode>> exprs_;
 };
 
+class IdExprNode : public ExprNode {
+ public:
+  IdExprNode(std::string id) : id_{id} {}
```
Comment (Lai-YT) [inline ast.cpp:60]:
```suggestion
  IdExprNode(const std::string& id) : id_{id} {}
```

String copying is considered an expensive operation and is unnecessary in this context.
Prefer passing `std::string`s as const-reference over value.

**Review by Lai-YT — CHANGES_REQUESTED** [review]

**Comment by leewei05** [comment]
@Lai-YT updated!

**Review by Lai-YT — APPROVED** [review]
LGTM 🎉

**Pull request merged**